### PR TITLE
Quarterdeck send invite route

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds a route to Quarterdeck for sending email invites to new users. When the endpoint is invoked, a user invitation record is added to the database and an email is sent to the email address in the request with a verification token.

Fixes SC-15467

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/db/models/users.go [hunk 1]
```diff
@@ -361,32 +363,56 @@ func (u *User) UpdateLastLogin(ctx context.Context) (err error) {
...
+		}
+		userID = ulids.New()
+	} else if _, err := GetOrgUser(ctx, user.ID, u.orgID); err != nil {
+		if !errors.Is(err, ErrNotFound) {
+			return nil, err
+		}
+		userID = user.ID
+	} else {
+		return nil, ErrUserOrgExists
 	}
```
Comment (bbengfort) [inline pkg/quarterdeck/db/models/users.go:400]: This was a little hard to read on the first pass; is this code also what you were expecting?

I'm actually not sure if the below code will work tbh -- but I am mostly trying to figure out how to interpret the if/elses. 

```suggestion
	} else if _, err := GetOrgUser(ctx, user.ID, u.orgID) {
		// Ensure the user hasn't already been added to the organization
		if err == nil {
			return nil, ErrUserOrgExists
		}
		
		if err != nil && !errors.Is(err, ErrNotFound) {
			return nil, err
		}
		
		// Assign the existing user's ID to the invitation
		userID = user.ID
	}
```

File: pkg/quarterdeck/db/models/users.go [hunk 2]
```diff
@@ -361,32 +363,56 @@ func (u *User) UpdateLastLogin(ctx context.Context) (err error) {
...
+		return nil, err
+	}
+
+	// Get the invitee's ID if they already exist or create a new ID
+	var userID ulid.ULID
+	if user, err := GetUserEmail(ctx, email, ulid.ULID{}); err != nil {
+		if !errors.Is(err, ErrNotFound) {
+			return nil, err
+		}
+		userID = ulids.New()
```
Comment (bbengfort) [inline pkg/quarterdeck/db/models/users.go:392]: Some comments might be helpful here; so if we try to fetch their email and `err == nil` then you're expecting us to go to the `else if` block? It is a bit tricky to read.

**Review by bbengfort — APPROVED** [review]
Looks good; mind having @eschmier take a look at the email copy?

File: pkg/quarterdeck/db/models/users.go (hunk 1 above)
Comment (pdeziel) [reply pkg/quarterdeck/db/models/users.go]: I attempted to break up the ifs a bit, let me know if that's clearer.

Comment (bbengfort) [reply pkg/quarterdeck/db/models/users.go]: definitely better; thanks!

**Pull request merged**